src/App.tsx

```diff
@@ -26,68 +26,87 @@ type ReorderOptions = {
   isReversed: boolean,
 };
 
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
 // Use this function in the render method to prepare goods
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  if (sortType) {
+    visibleGoods.sort((goodA, goodB) => (sortType === SortType.ALPHABET
+      ? goodA.localeCompare(goodB)
+      : goodB.length - goodA.length));
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
 export const App: React.FC = () => {
+  const [sortState, setSortState] = useState<State>({
+    sortType: SortType.NONE,
+    isReversed: false,
+  });
+
+  const onSortClick = (newSortType: keyof typeof SortType) => setSortState(
+    prev => ({ ...prev, sortType: SortType[newSortType] }),
+  );
+
+  const onReversClick = () => setSortState(prev => (
+    { ...prev, isReversed: !prev.isReversed }));
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
           className="button is-info is-light"
+          onClick={() => onSortClick('ALPHABET')}
```

Comment:
Use enum value instead of string 'ALPHABET' or create a separate handlers and use them like:

```
  sortByAlphabet = () => {
    this.setState({ sortType: SortType.ALPHABET });
  };
```